File: app/src/main/java/com/vocaby/app/ui/SearchResultsBodyFragment.java
package com.vocaby.app.ui;

import android.content.Context;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import androidx.viewpager2.adapter.FragmentStateAdapter;

import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.vocaby.app.Constants;
import com.vocaby.app.R;
import com.vocaby.app.adapters.DefinitionsAdapter;
import com.vocaby.app.models.EntryModel;

public class SearchResultsBodyFragment extends Fragment {
    private Context ctx;
    private TextView word;
    private TextView pronunciation;
    private DefinitionsAdapter adapter;

    private static final String ENTRY_DATA_PARAM = "entryData";

    private EntryModel entryData;

    public SearchResultsBodyFragment() {
        // Required empty public constructor
    }
    public static SearchResultsBodyFragment newInstance(EntryModel entryData) {
        SearchResultsBodyFragment fragment = new SearchResultsBodyFragment();
        Bundle args = new Bundle();
        args.putParcelable(ENTRY_DATA_PARAM, entryData);
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getArguments() != null) {
            entryData = getArguments().getParcelable(ENTRY_DATA_PARAM);
        }

        ctx = requireActivity().getApplicationContext();
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View view = inflater.inflate(R.layout.fragment_search_results_body, container, false);
        word = view.findViewById(R.id.word_header);
        pronunciation = view.findViewById(R.id.pronunciation);

        RecyclerView recyclerView = view.findViewById(R.id.definitions_recycler_container);
        recyclerView.setEnabled(false);
        adapter = new DefinitionsAdapter(ctx);
        recyclerView.setAdapter(adapter);
        recyclerView.setLayoutManager(new LinearLayoutManager(ctx));

        if (entryData.isEmpty()) {
            populateNoDefinition();
        } else {
            populateView(entryData);
            adapter.setWordData(entryData);
        }

        return view;
    }

    private void populateView(EntryModel entryData) {
        word.setText(entryData.getEntry());
        String pronunciationText = entryData.getPronunciation().replaceAll("\n","");
        if(!pronunciationText.isEmpty()) {
            pronunciation.setVisibility(View.VISIBLE);
            pronunciation.setText(pronunciationText);
        }
    }

    private void populateNoDefinition() {
        word.setText(getResources().getString(R.string.no_definition_found));
        pronunciation.setVisibility(View.GONE);
    }
}

File: app/src/main/java/com/vocaby/app/ui/dictionary/SearchResultsBodyFragment.kt
package com.vocaby.app.ui.dictionary

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.DividerItemDecoration
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.vocaby.app.R
import com.vocaby.app.adapters.DefinitionsAdapter
import com.vocaby.app.models.dictionary.EntryModel

class SearchResultsBodyFragment : Fragment() {
    private lateinit var ctx: Context
    private var entryData: EntryModel? = null
    private lateinit var header: TextView
    private lateinit var pronunciation: TextView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        if (arguments != null) {
            entryData = requireArguments().getParcelable(ENTRY_DATA_PARAM)
        }

        ctx = requireActivity().applicationContext
    }

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        // Inflate the layout for this fragment
        val view = inflater.inflate(R.layout.fragment_search_results_body, container, false)
        header = view.findViewById(R.id.word_header)
        pronunciation = view.findViewById(R.id.pronunciation)

        entryData?.let { data ->
            val recyclerView: RecyclerView = view.findViewById(R.id.definitions_recycler_container)
            val definitionsAdapter = DefinitionsAdapter(ctx)
            definitionsAdapter.setWordData(data)

            recyclerView.apply {
                isEnabled = false
                adapter = definitionsAdapter
                layoutManager = LinearLayoutManager(ctx)

                if (data.definitionGroups.size > 1)
                    recyclerView.addItemDecoration(DividerItemDecoration(ctx, LinearLayoutManager.VERTICAL))
            }

            populateView(data)
        } ?: populateNoDefinition()

        return view
    }

    private fun populateView(entryData: EntryModel) {
        header.text = entryData.entry
        if (!entryData.pronunciation.isNullOrEmpty()) {
            pronunciation.visibility = View.VISIBLE
            pronunciation.text = entryData.pronunciation
        }
    }

    private fun populateNoDefinition() {
        header.text = resources.getString(R.string.no_definition_found)
        pronunciation.visibility = View.GONE
    }

    companion object {
        private const val ENTRY_DATA_PARAM = "entryData"
        fun newInstance(entryData: EntryModel?): SearchResultsBodyFragment {
            val fragment = SearchResultsBodyFragment()
            val args = Bundle()
            args.putParcelable(ENTRY_DATA_PARAM, entryData)
            fragment.arguments = args
            return fragment
        }
    }
}